src/test/ui/closures/2229_closure_analysis/by_value.rs

```diff
@@ -0,0 +1,39 @@
+// Test that we handle derferences properly when only some of the captures are being moved with
+// `capture_disjoint_fields` enabled.
```

Comment:
Can we make this comment more specific, or add comments below?

Reply:
e.g., test that move closures only capture data from the enclosing stack frame and not the deref of a box